integration/cri-containerd/test.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+# 
+# Copyright (c) 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -x
+set -eE
+
+err_report() {
+	cat /tmp/test-cri/cri-containerd.log
+	cat /tmp/test-cri/containerd.log
+}
+
+trap 'err_report' ERR
+
+
+repo="github.com/containerd/cri-containerd"
+# FIXME: cri-containerd fails using macvtap.
+sudo sed -r -i -e 's/^(internetworking_model=).*/\1"bridged"/g' /usr/share/defaults/clear-containers/configuration.toml
+
+CRITEST=${GOPATH}/bin/critest
+# make sure cri-containerd test install the proper installation for testing
+rm -f "${CRITEST}"
+pushd ${GOPATH}/src/${repo}
+
+FOCUS="runtime should support basic operations on container" CRI_CONTAINERD_FLAGS="--containerd-runtime-engine cc-runtime --stream-port 10011" make test-cri
+
+passing_test=(
```

Comment:
This is great. However, how about running all the tests and asserting that this list pass, and all the other tests fail?

Assuming we update `versions.txt` to pin the version of `cri-containerd` we're using, this should work as every time we update that version, and re-run the tests locally we *might* end up getting more tests passing, which can then be added to this list.

go1.9 provides a `go test -list '.*'` command which is supposed to list all tests in a package, but I cannot get it to work locally. As an alternative, this seems to be a more reliable method of listing all test files and test functions:

```
find . -type f|grep -v vendor/|grep "_test.go"|xargs grep "^func Test[^ ]"|sed 's/func //g'|cut -d\( -f1|cut -d: -f2|sort
```